guide/locations/_AWS.md

```diff
@@ -43,6 +43,51 @@ Below are examples of configuration options that use values specific to AWS EC2:
   For example, `securityGroups: mygroup1` or `securityGroups: [ mygroup1, mygroup2 ]`.
 
 
+### Using a Registered Key Pair
+
+You can specify a `keyPair` to use for initial provisioning as a configuration option.
+If this is omitted, Brooklyn will use jclouds to create a new ad hoc key pair at AWS
+for that machine, and it will delete it afterwards.  This is usually seamless and
+occurs behind the scenes, with the post-provision user set up and configured as normal
+for all locations.  However if you are optimizing for creation of many machines,
+or you are using an image where keypair installation is not reliable (such as [some
+RightImages](https://issues.apache.org/jira/browse/JCLOUDS-1356)), then you may wish
+to use a registered key pair.
+
+First, in the AWS Console, open the EC2 service in the region you are interested in,
+then click "Key Pairs" at the left.  For `us-east-1`, the link is 
+[here](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#KeyPairs:sort=keyName).
+Click "Create Key Pair" (or "Import Key Pair" if you want to provide a public key) and
+follow the instructions.
+
+Then define your location as follows for `aws-us-east-1`.  Make sure to replace the
+`XXXX` sections with the key-pair name defined above and the corresponding private key data. 
+
+```yaml
+brooklyn.catalog:
+  version: "1.0"
+  itemType: location
+  items:
+  - id: aws-base
+    item:
+      type: jclouds:aws-ec2
+      brooklyn.config:
+        identity: XXXXXXXXXXXXXXXX
+        credential: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXx
+  - id: aws-us-east-1
+    item:
+      type: aws-base
+      region: us-east-1
+      keyPair: XXXXXXXXX
+      login.privateKeyData: |
```

Comment:
isn't `loginUser.privateKeyData`? there is also `loginUser.privateKeyFile` which can maybe be more convenient in some cases.

Reply:
you are completely right @andreaturli - so how the heck is my setup working with `login.privateKeyData` ???

Reply:
aha i have `brooklyn.location.jclouds.extraSshPublicKeyData` set so i think it is using `TemplateOptions.authorizePublicKey` and then using my normal `id_rsa`.  let me disable that and triple check these instructions are right.  TY!

Reply:
no, it wasn't that, it's that config in location defns need to be in `brooklyn.config` -- updating this